### PR TITLE
BaseBridgeReceiver unit tests

### PR DESCRIPTION
**Pull request by scott-silver**

### CODE REVIEW

File: test/bridges/base-bridge-receiver-test.ts [hunk 1]
```diff
@@ -0,0 +1,452 @@
+import { ethers, event, expect, wait } from './../helpers';
+import { utils } from 'ethers';
+import {
+  BaseBridgeReceiverHarness__factory,
+  Timelock__factory
+} from '../../build/types';
+
+const TYPES = ['address[]', 'uint256[]', 'string[]', 'bytes[]'];
```
Comment (kevincheng96) [inline test/bridges/base-bridge-receiver-test.ts:8]: nit: `BRIDGE_RECEIVER_CALLDATA_ABI`?

File: test/bridges/base-bridge-receiver-test.ts [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+      govTimelock,
+      localTimelock
+    } = await makeBridgeReceiver({initialize: true});
+
+    await expect(
+      baseBridgeReceiver.initialize(govTimelock.address, localTimelock.address)
+    ).to.be.revertedWith("custom error 'AlreadyInitialized()'");
+  });
+
+  it('acceptLocalTimelockAdmin reverts if not called by localTimelock', async () => {
```
Comment (kevincheng96) [inline test/bridges/base-bridge-receiver-test.ts:100]: nit: test name is inconsistent with the others, could be `acceptLocalTimelockAdmin > revert for unauthorized caller` as well

File: test/bridges/base-bridge-receiver-test.ts [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+    const { baseBridgeReceiver } = await makeBridgeReceiver();
+
+    await expect(
+      baseBridgeReceiver.acceptLocalTimelockAdmin()
+    ).to.be.revertedWith("custom error 'Unauthorized()'");
+  });
+
+  it('setLocalTimelock > reverts for unuauthorized caller', async () => {
```
Comment (kevincheng96) [inline test/bridges/base-bridge-receiver-test.ts:108]: nit: typo in unauthorized

File: test/bridges/base-bridge-receiver-test.ts [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+  });
+
+  it('setGovTimelock > sets gov timelock', async () => {
+    const {
+      baseBridgeReceiver,
+      govTimelock,
+      signers
+    } = await makeBridgeReceiver({ initialize: false });
+
+    const [localTimelockSigner, newGovTimelockSigner] = signers;
```
Comment (kevincheng96) [inline test/bridges/base-bridge-receiver-test.ts:167]: `govTimelcok` and `localTimelock` are also derived from signers, so probably safer to use some other hard-coded address for the `newGovTimelock`

File: test/bridges/base-bridge-receiver-test.ts [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+
+async function makeBridgeReceiver({ initialize } = { initialize: true }) {
+  const [_defaultSigner, govTimelockAdmin, ...signers] = await ethers.getSigners();
+
+  const BaseBridgeReceiverFactory = (await ethers.getContractFactory('BaseBridgeReceiverHarness')) as BaseBridgeReceiverHarness__factory;
+  const baseBridgeReceiver = await BaseBridgeReceiverFactory.deploy();
+  await baseBridgeReceiver.deployed();
+
+  const govTimelock = await makeTimelock({ admin: govTimelockAdmin.address });
+  const localTimelock = await makeTimelock({ admin: baseBridgeReceiver.address });
```
Comment (kevincheng96) [inline test/bridges/base-bridge-receiver-test.ts:37]: This doesn't look right

File: test/bridges/base-bridge-receiver-test.ts (hunk 4 above)
Comment (kevincheng96) [reply test/bridges/base-bridge-receiver-test.ts:167]: Or maybe you could do it the other way around and use some hard-coded address for the original `govTimelock` and `localTimelock` instead.

File: test/bridges/base-bridge-receiver-test.ts (hunk 5 above)
Comment (kevincheng96) [reply test/bridges/base-bridge-receiver-test.ts:37]: Oops nevermind. Misread it!

File: test/bridges/base-bridge-receiver-test.ts (hunk 4 above)
Comment (kevincheng96) [reply test/bridges/base-bridge-receiver-test.ts:167]: Disregard this. I misread the setup code

File: test/bridges/base-bridge-receiver-test.ts [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+  it('state > reverts if proposal id is greater than proposalCount', async () => {
+    const { baseBridgeReceiver } = await makeBridgeReceiver();
+
+    const proposalCount = await baseBridgeReceiver.proposalCount();
+
+    await expect(
+      baseBridgeReceiver.executeProposal(proposalCount.add(1))
+    ).to.be.revertedWith("custom error 'InvalidProposalId()'");
+  });
+});
```
Comment (kevincheng96) [inline test/bridges/base-bridge-receiver-test.ts:452]: We should also test the case where a transaction is already queued.

**Review by kevincheng96 — APPROVED** [review]
Sweet, looks good overall! I think it's just missing the case where a transaction is already queued, but seems very comprehensive.

File: test/bridges/base-bridge-receiver-test.ts (hunk 6 above)
Comment (scott-silver) [reply test/bridges/base-bridge-receiver-test.ts:452]: Is that covered by: `processMessage > reverts for repeated transactions`?

File: test/bridges/base-bridge-receiver-test.ts (hunk 1 above)
Comment (scott-silver) [reply test/bridges/base-bridge-receiver-test.ts]: i like it; updated

File: test/bridges/base-bridge-receiver-test.ts (hunk 2 above)
Comment (scott-silver) [reply test/bridges/base-bridge-receiver-test.ts]: updated

File: test/bridges/base-bridge-receiver-test.ts (hunk 6 above)
Comment (kevincheng96) [reply test/bridges/base-bridge-receiver-test.ts:452]: Ah I missed it. Nice work!

**Pull request merged**